### ISSUE TITLE
Completion block called twice on imageAtURL message

### ISSUE THREAD

**Issue by GuyFran** [MugunthKumar/MKNetworkKit#82]
Hey.
I am trying to figure out why my completion is called twice.

Here's a sample of my code
- (void)viewWillAppear:(BOOL)animated
  {
  [super viewWillAppear:animated];
  [self updatemyimage];
  }

-(void) updatemyimage
{
NSLog("here");
self.imageLoadingOperation = [self.engine imageAtURL:[NSURL URLWithString:my_pic_url] onCompletion:^(UIImage *fetchedImage, NSURL *url, BOOL isInCache)
    {
      if([my_pic_url isEqualToString:[url absoluteString]])
        {
           if (isInCache) {
               NSLog(@"Cache");
           }
           else {
               NSLog(@"Network");
           }
                //Do the stuff
        }
    }];
    [self.engine enqueueOperation:imageLoadingOperation];
}

This works but outputs TWICE "Cache" or "Network"  so i'm afraid the picture is downloaded twice (for now I'n on wifi with a small pic, so I can't really tell, but that's my logical guess).

I know that for each "Refresh" of my main view, I only fire the updatemyimage message only once since I only have ONE output for "here".

Another issue I have is I don't really understand the logic behind the if condition "my_pic_url isEqualToString:[url absoluteString]". Since there's one instance of operation for each  GET request (here, my picture), it seems logical that the very same operation will received the data of the url specified in the same operation ... or did I miss something ?
